realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -396,15 +402,17 @@ private static void initializeSyncedRealm(Realm realm) {
 
             // Assumption: When SyncConfiguration then additive schema update mode.
             final OsRealmSchema schema = new OsRealmSchema(schemaCreator);
-            long newVersion = realm.configuration.getSchemaVersion();
+            long newVersion = configuration.getSchemaVersion();
```

Comment:
Don't you need to call `schema.close()` somewhere? it seems the native resoure created in `OsRealmSchema` will be leaked?